docs/fundamentals/code-analysis/quality-rules/ca1418.md

```diff
@@ -0,0 +1,121 @@
+
+## Rule description
+
+The platform compatibility attributes derived from <xref:System.Runtime.Versioning.OSPlatformAttribute> use string literals for OS platform names with optional version part. The string should consist a **known** platform name and have no version or with a **valid** version part.
+
+The **known** platform names list is populated from two places:
+
+- `PlatformName` part of <xref:System.OperatingSystem> guard methods `OperatingSystem.Is(PlatformName)[VersionAtLeast]()`, for example guard method <xref:System.OperatingSystem.IsWindows?displayProperty=nameWithType> would make the `Windows` part of the known platforms.
+- The project's MSBuild item group of `SupportedPlatform`s, including the default [MSBuild SupportedPlatforms list](https://github.com/dotnet/sdk/blob/main/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.SupportedPlatforms.props). This is the project specific knowledge of known platforms, this allows class library authors to add more platforms into the known platforms list:
+
+   ```XML
+    <ItemGroup>
+      <SupportedPlatform Include="PlatformName" />
+    </ItemGroup>
+  ```
+
+If the platform string contains **version** part it should be a **valid** <xref:System.Version> with a format: `major.minor[.build[.revision]]`.
+
+### Violations
+
+- `Solaris` is an **unknown** platform name because it is **not** included in the default [MSBuild SupportedPlatforms list](https://github.com/dotnet/sdk/blob/main/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.SupportedPlatforms.props) and there is **no** guard method like `OperatingSystem.IsSolaris()`in <xref:System.OperatingSystem> class.
+
+  ```csharp
+  [SupportedOSPlatform("Solaris")] // warns: The platform 'Solaris' is not a known platform name
+  public void SolarisApi() { }
+  ```
+
+- `Android` is a **known** platform because there is <xref:System.OperatingSystem.IsAndroid?displayProperty=nameWithType> guard method exist in <xref:System.OperatingSystem> type, but the version part is not valid version, it should have at least two integers separated by dot.
+
+  ```csharp
+  [UnsupportedOSPlatform("Android10")] // warns: Version '10' is not valid for platform 'Android'. Use a version with 2-4 parts for this platform.
+  public void DoesNotWorkOnAndroid() { }
+  ```
+
+- `Linux` is a **known** platform because it is included in the default [MSBuild SupportedPlatforms list](https://github.com/dotnet/sdk/blob/main/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.SupportedPlatforms.props), also a guard <xref:System.OperatingSystem.IsLinux?displayProperty=nameWithType> exist, but there is **no** any versioned guard like `System.OperatingSystem.IsLinuxVersionAtLeast(int,int)` exist for Linux platform, therefore no version is supported on Linux.
```

Comment:
```suggestion
- `Linux` is a **known** platform because it is included in the default [MSBuild SupportedPlatforms list](https://github.com/dotnet/sdk/blob/main/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.SupportedPlatforms.props), and there's also a guard method named <xref:System.OperatingSystem.IsLinux?displayProperty=nameWithType>. However, there are no versioned guard methods like `System.OperatingSystem.IsLinuxVersionAtLeast(int,int)` for the `Linux` platform, therefore no version part is supported on Linux.
```